dimod/constrained.py

```diff
@@ -35,6 +36,7 @@
 
 import numpy as np
 
+import dimod
```

Comment:
I don't think you need this, you use it for `dimod.QuadraticModel`, but that's already imported lower down